share/picongpu/tests/CurrentDeposition/lib/python/test/CurrentDeposition/assignment_and_W_func.py

```diff
@@ -0,0 +1,116 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2023 PIConGPU contributors
+Authors: Hannes Wolf
+License: GPLv3+
+
+Implements the assignment functions of order 0 to 3 as well as a function to
+calculate the current deposition vector.
+
+"""
+
+
+# functions for single values
+def NGP(x):
+    """Generates hat function with width 1 (symmetrical around x=0) -
+    assignment function of order 0 for single values x.
+
+    Parameters:
+    x (float): distance between particle and evaluation point
+
+    Returns:
+    y (float): value of the hat function NGP(x)
+
+    """
+
+    if abs(x) < 1/2:
+        y = 1
+    elif abs(x) == 1/2:
+        y = 1/2
+    else:
+        y = 0
+    return y
+
+
+def CIC(x):
+    """Generates triangle with width 2 (symmetrical around x=0) -
+    assignment function of order 1 for single values x.
+
+    Parameters:
+    x (float): distance between particle and evaluation point
+
+    Returns:
+    y (float): value of the triangle function CIC(x)
+
+    """
+
+    if abs(x) < 1:
+        y = 1 - abs(x)
+    else:
+        y = 0
+    return y
+
+
+def TSC(x):
+    """assignment function of order 2 for single values x.
+
+    Parameters:
+    x (float): distance between particle and evaluation point
+
+    Returns:
+    y (float): value of the function TSC(x)
+
+    """
+
+    if abs(x) < 3/2 and abs(x) >= 1/2:
+        y = ((3/2 - abs(x))**2)/2
+    elif abs(x) < 1/2:
+        y = (-(x**2) + 3/4)
+    else:
+        y = 0
+    return y
+
+
+def PQS(x):
+    """assignment function of order 3 for single values x.
+
+    Parameters:
+    x (float): distance between particle and evaluation point
+
+    Returns:
+    y (float): value of the function PQS(x)
+
+    """
+    if abs(x) < 2 and abs(x) >= 1:
+        y = 1/6*(2 - abs(x))**3
+    elif abs(x) < 1:
+        y = (2/3 - x**2 + (abs(x)**3)/2)
+    else:
+        y = 0
+    return y
+
+
+def W(s1, s2, s3, s4, s5, s6):
+    """Calculation of the current deposition (one component).
+
+    Parameters:
+    s1, s2, ..., s6(float): values of the assignment function at the old and
+                            the new coordinates
+    # the variables are just placeholders in this case. The values

```

Comment:
I think conventionally this form of explanation would go above the `Parameters` section as part of the long description but I don't have a strong opinion here. I might just render badly in some IDEs that try to be clever about such things.